ros2bag_tools/ros2bag_tools/exporter/image.py

```diff
@@ -43,39 +135,45 @@ def add_arguments(parser):
         parser.add_argument('--name', default='%t.png',
                             help="""Filename pattern of output images.
                             Placeholders:
-                                %tpc ... topic
-                                %t   ... timestamp
-                                %i   ... sequence index""")
+                                %%tpc ... topic
+                                %%t   ... timestamp
+                                %%i   ... sequence index""")
         parser.add_argument('--encoding', default='passthrough',
                             help='Output image encoding')
         parser.add_argument('--demosaicing', choices=['linear', 'vng', 'ea'],
                             help='Bayer pattern demosaicing algorithm')
+        parser.add_argument('--input-encoding', default='',
```

Comment:
I understand the use case for this. I think to avoid surprises when the overwritten encoding mismatches the data, it would be helpful to read that in error output.

For example, the user may set --input-encoding mono8 for an export of a color image.
@devrite do you know what the user would currently see as output? does the output show that the argument is wrong?

Reply:
Since image export will try to do an conversion if output enc and input enc mismatch, this is no issue. But I will add a verification that the stored input encoding and the desired input encoding (override) will have the same output size. Otherwise it is likely wrong or not will not work anyway.

So if eg., input is mono8 but should be treated as bayer_rggb8 both will have one channel with uint8 as element and the resulting size per pixel will be the same. If this condition does not hold I will raise an exception.

Reply:
Sounds good!